samples/mixed/provider/src/main.rs

```diff
@@ -180,6 +180,11 @@ async fn publish(subscription_map: Arc<Mutex<SubscriptionMap>>, entity_id: &str,
     }
 }
 
+/// Starts the vehicle simulator.
+///
+/// # Arguments
+/// * `subscription_map` - Subscription map.
+/// * `vehicle` - Vehicle.
```

Comment:
can we add a little more detail to these comments? they're adding no value on top of the function and parameter names. maybe @ashbeitz can help since I think he wrote these

Reply:
Added extra details to the vehicle argument